tests/test_custom_checkers/test_inconsistent_returns.py

```diff
@@ -153,3 +154,84 @@ def func():
             ignore_position=True,
         ):
             self.checker.visit_functiondef(func_node)
+
+    def test_z3_unfeasible_inconsistent_return(self):
+        src = """
+        def func(x: int) -> int:
+            '''
+            Preconditions:
+                - x > 5
+            '''
+            if x < 0:
+                return
+            return x
+        """
+        z3v = Z3Visitor()
+        mod = z3v.visitor.visit(astroid.parse(src))
+        mod.accept(CFGVisitor())
+        func_node = next(mod.nodes_of_class(nodes.FunctionDef))
+        inconsistent_return_node, _ = mod.nodes_of_class(nodes.Return)
+
+        with self.assertAddsMessages(
+            pylint.testutils.MessageTest(
+                msg_id="inconsistent-returns",
+                node=inconsistent_return_node,
+            ),
+            ignore_position=True,
+        ):
+            self.checker.linter.config.z3 = True
```

Comment:
In general, rather than set this attribute manually, you should create a new *test suite class*, and then set the `z3: True` configuration value as a class variable in the test suite---see how other tests set config values. The new tests you're adding should be put under these new test suite classes that have `z3` enabled.

Reply:
This comment hasn't been resolved